tests/unit/query_graph/on_demand_function/test_binary.py

```diff
@@ -0,0 +1,170 @@
+"""
+Test the binary nodes in the on-demand view code generation.
+"""
+import numpy as np
+import pandas as pd
+import pytest
+
+from featurebyte.enum import DBVarType
+from featurebyte.query_graph.node.base import BinaryOpWithBoolOutputNode
+from featurebyte.query_graph.node.binary import (
+    AddNode,
+    AndNode,
+    DivideNode,
+    EqualNode,
+    GreaterEqualNode,
+    GreaterThanNode,
+    IsInNode,
+    LessEqualNode,
+    LessThanNode,
+    ModuloNode,
+    MultiplyNode,
+    NotEqualNode,
+    OrNode,
+    PowerNode,
+    SubtractNode,
+)
+from featurebyte.query_graph.node.metadata.config import (
+    OnDemandFunctionCodeGenConfig,
+    OnDemandViewCodeGenConfig,
+)
+from featurebyte.query_graph.node.metadata.sdk_code import VariableNameGenerator, VariableNameStr
+
+NODE_PARAMS = {"name": "node_name", "parameters": {"value": None}}
+
+
+def evaluate_and_compare_result(feat1, feat2, odfv_expr, odff_expr):
+    """Evaluate the odfv & odff expressions & compare the results"""
+    # check the odfv expression can be evaluated
+    out_odfv = eval(odfv_expr)
+
+    # check the odff expression can be evaluated
+    out_vals = []
+    for feat1, feat2 in zip(feat1, feat2):
+        _ = feat1, feat2
+        out_vals.append(eval(odff_expr))
+    out_odff = pd.Series(out_vals)
+
+    # check the consistency between two expressions
+    pd.testing.assert_series_equal(out_odfv, out_odff)
```

Comment:
This function is used to check the output generated from on demand feature view function (for feast) is consistent with on demand feature function (for DataBricks).